server/common/config/external_config.py

```diff
@@ -0,0 +1,103 @@
+import os
+
+from server.common.config.base_config import BaseConfig
+from server.common.errors import ConfigurationError
+from server.common.config import get_secret_key
+from server.common.errors import SecretKeyRetrievalError
+
+
+class ExternalConfig(BaseConfig):
+    """Manages the config attribute associated with external configuration sources, such as
+    environment variables or the AWS Secrets Manager."""
+
+    def __init__(self, app_config, default_config):
+        super().__init__(app_config, default_config)
+        try:
+            self.environment = default_config["environment"]
+            self.aws_secrets_manager__region = default_config["aws_secrets_manager"]["region"]
+            self.aws_secrets_manager__secrets = default_config["aws_secrets_manager"]["secrets"]
+
+        except KeyError as e:
+            raise ConfigurationError(f"Unexpected config: {str(e)}")
+
+    @staticmethod
+    def convert_value(value: str):
```

Comment:
To keep this class as clean as possible, I'd move this to the `util` directory. Probably `type_conversion_utils.py` seems like a good fit.

Reply:
Also should this be hardened to ensure no issues with capitalization (TRUE or FALSE)?

Reply:
I can add TRUE and FALSE.  I was mainly trying to hit the python and yaml casing.  But I can make this case insensitive.